### ISSUE TITLE
[Due for payment 2026-03-04] Screen Reader: Many Pages: The displayed error message is not announced

### ISSUE THREAD

**Issue by olenamihno** [Expensify/App#77572]
## Action Performed:
Prerequisites:
- the user is not signed in

1. Using Android TalkBack or iOS VoiceOver, open the Expensify app
2. Swipe to the 'Continue' button and double-tap to activate without entering the phone or email
3. Note if the displayed error message is announced.

Other Occurrences:
1. On Sign In - Enter Code
2. Onboarding: all fields in the flow
3. On Inbox: the field 'Enter your boss's email'
4. On Profile - Display Name
5. On Profile - Date of Birth
6. On Profile - Address
7. On Profile - Legal Name
8. On Profile - Verify Email
9. On Security - Merge Accounts
10. On Security - Merge Accounts - Enter Code
11. On Security - Close Account
12. On Wallet - Add Bank Account
13. On Create Expense - Confirm Details
14. On Save the World - I know a teacher
15. On Create Workspace

## Expected Result:
The displayed error message should be announced once it appears.

## Actual Result:
When a user triggers an error, the error message appears visually, but is not announced.

## Area issue was found in:
Sign in

## Failed WCAG checkpoints
WCAG 4.1.3 Status Messages
https://www.w3.org/WAI/WCAG22/Understanding/status-messages

## User impact:
When appropriate roles or properties are assigned to status messages, the new content is spoken by screen readers in such a way as to assist blind and low vision users. Assigning proper roles or properties to status messages provides possible future uses and personalization opportunities, such as the potential to be exploited by assistive technologies created for users with some cognitive disabilities.

## Suggested resolution:
Ensure the error message is being announced once it appears.
https://appt.org/en/docs/react-native/samples/accessibility-live-region

## Environment:
Samsung Galaxy S24 Ultra - Android 14, Partner - Native (No mobile browser)

## Screenshots/Videos

![Image](https://github.com/user-attachments/assets/ff60c25f-66d3-479d-ba0d-f333c4ad3d8a)

https://github.com/user-attachments/assets/c26be82d-2fa3-4a72-ab82-4bdea1660a36

**Version Number:** Production
**Reproducible in staging?:** NA
**Reproducible in production?:** Yes
**Email or phone of affected tester (no customers):** No
**Logs:** https://stackoverflow.com/c/expensify/questions/4856  NA
**Notes/Photos/Videos:** Any additional supporting documentation  Yes
**Expensify/Expensify Issue URL:** https://platform.applause.com/company/12102/products/36628/community-issues/7004827
**Issue reported by:** NA
**Slack conversation:** NA

<details><summary>Issue Owner</summary>Current Issue Owner: @</details>

**Comment by suneox** [comment]
Overal @marufsharifi proposal to add `accessibilityLiveRegion` works on a real mobile device — LGTM. Based on MevinBot review, the suggested using `role` instead of `accessibilityRole` differs from the current codebase, so we can to address and update that later during the PR review.

🎀 👀 🎀 C+ reviewed
